src/codec.rs

```diff
@@ -109,6 +109,20 @@ impl<W: ::std::io::Write> Output for W {
 	}
 }
 
+struct ArrayVecWrapper<T: arrayvec::Array>(ArrayVec<T>);
+
+impl<T: arrayvec::Array<Item=u8>> Output for ArrayVecWrapper<T> {
```

Comment:
Why do we need this wrapper? Can we not directly implement `Output` on `ArrayVec`?

Reply:
no because it will conflict with `impl<W: ::std::io::Write> Output for W` if ArrayVec implement Write in the future